### PR TITLE
PEP8 contamination and visibility calculator

### PR DESCRIPTION
**Pull request by hover2pi**
This nightmare is almost over. Please merge this pull request which makes the contamination and visibility calculator PEP8 compliant. 

`Showing 11 changed files with 1,744 additions and 1,463 deletions.`

### CODE REVIEW

**Comment by julesfowler** [comment]
Oh boy. Let's goooo.

File: ExoCTK/contam_visibility/quaternionx.py [hunk 1]
```diff
@@ -1,292 +1,328 @@
...
+        QV = Qmake_a_point(V)
+        QV = self * QV * self.conjugate()
+        return Vector(QV.q1, QV.q2, QV.q3)
+
+    def inv_cnvrt(self, V):
+        """Rotates a vector from the ending frame to the starting frame
+        defined by the Q"""
+        QV = Qmake_a_point(V)
+        QV = self.conjugate() * QV * self
+        return Vector(QV.q1, QV.q2, QV.q3)
```
Comment (julesfowler) [inline ExoCTK/contam_visibility/quaternionx.py]: Again I think we can spell out `convert`.

File: ExoCTK/contam_visibility/quaternionx.py [hunk 2]
```diff
@@ -658,11 +715,13 @@ def rotate_about_eigenaxis(self, angle, eigenaxis):
         R32 = eigenaxis.y * eigenaxis.z * (1 - cos_ang) + eigenaxis.x * sin_ang
         R33 = cos_ang + eigenaxis.z**2 * (1 - cos_ang)
```
Comment (julesfowler) [inline ExoCTK/contam_visibility/quaternionx.py]: These aren't global variables -- they shouldn't be capitalized.

File: ExoCTK/contam_visibility/sossContamFig.py [hunk 3]
```diff
@@ -2,18 +2,19 @@
...
 from . import visibilityPA as vpa
 import os
+import sys
 import pkg_resources
-from bokeh.io import show
 from bokeh.layouts import gridplot
 from bokeh.plotting import figure
-from bokeh.models import Range1d, LinearColorMapper, LogColorMapper, LogTicker, ColorBar, Label
+from bokeh.models import Range1d, LinearColorMapper, Label
 from bokeh.palettes import inferno
```
Comment (julesfowler) [inline ExoCTK/contam_visibility/sossContamFig.py]: Imports are out of order.  
Should be: 
```
import os
import pkg_resources
import sys

from astropy.io import fits
from bokeh.io import show
from bokeh.layouts import gridplot
from bokeh.models import Lable, LinearColorMapper, Range1d
from bokeh.palettes import inferno

from . import visibilityPA as vpa
```

File: ExoCTK/contam_visibility/sossContamFig.py [hunk 4]
```diff
@@ -42,257 +43,315 @@ def contam(cube, targetName='noName', paRange=[0,360], badPA=[], tmpDir="", fig=
...
-
-    #start calculations
-    lam_file = pkg_resources.resource_filename('ExoCTK', 'data/contam_visibility/lambda_order1-2.txt')
+
+    trace2dO1 = cube[0, :, :]  # order 1
+    trace2dO2 = cube[1, :, :]  # order 2
+    cube = cube[2:, :, :]  # all the angles
+
+    plotPAmin, plotPAmax = paRange
+    # suffix = '_PA'+str(plotPAmin)+'-'+str(plotPAmax)
```
Comment (julesfowler) [inline ExoCTK/contam_visibility/sossContamFig.py]: Remove this commented line.

File: ExoCTK/contam_visibility/sossContamFig.py [hunk 5]
```diff
@@ -42,257 +43,315 @@ def contam(cube, targetName='noName', paRange=[0,360], badPA=[], tmpDir="", fig=
...
-        TOOLS = 'pan,box_zoom,crosshair,reset,hover,save'
-        
-        y=np.array([0.,0.])
-        y1=0.07
-        y2=0.12
-        y3=0.17
-        y4=0.23
-        bad_PA_color = '#dddddd'
+
+        TOOLS = 'pan, box_zoom, crosshair, reset, hover, save'
```
Comment (julesfowler) [inline ExoCTK/contam_visibility/sossContamFig.py]: I know this is often a `bokeh` standard but it's still not a global variable.

File: ExoCTK/contam_visibility/sossContamFig.py [hunk 6]
```diff
@@ -42,257 +43,315 @@ def contam(cube, targetName='noName', paRange=[0,360], badPA=[], tmpDir="", fig=
...
+        s1.add_layout(Label(x=l.mean(), y=y3+0.02, x_units='data',
+                      y_units='data', text='K', render_mode='css',
+                      text_font_size='8pt'))
+
+        l = np.array([1.5168])
+        s1.line(l[0], [y3, y3+0.02], line_color='black')
+        s1.add_layout(Label(x=l.mean(), y=y3+0.02, x_units='data',
+                      y_units='data', text='K', render_mode='css',
+                      text_font_size='8pt'))
         #
```
Comment (julesfowler) [inline ExoCTK/contam_visibility/sossContamFig.py]: Should this empty commented line be here?

File: ExoCTK/contam_visibility/sossFieldSim.py [hunk 7]
```diff
@@ -1,200 +1,223 @@
 from astroquery.irsa import Irsa
-from astropy.io import fits
 from scipy.io import readsav
 import matplotlib.pyplot as plt
 import astropy.coordinates as crd
 import astropy.units as u
 import numpy as np
 import os
 import glob
-import pdb
```
Comment (julesfowler) [inline ExoCTK/contam_visibility/sossFieldSim.py]: I am a broken record -- please alphabetize these imports.

File: ExoCTK/contam_visibility/sossFieldSim.py [hunk 8]
```diff
@@ -1,200 +1,223 @@
...
+    modelParam = readsav(os.path.join(idlsave_path, 'modelsInfo.sav'),
+                         verbose=False)
+    models = modelParam['models']
+    modelPadX = modelParam['modelpadx']
+    modelPadY = modelParam['modelpady']
+    dimXmod = modelParam['dimxmod']
+    dimYmod = modelParam['dimymod']
+    jhMod = modelParam['jhmod']
+    hkMod = modelParam['hkmod']
+    teffMod = modelParam['teffmod']
```
Comment (julesfowler) [inline ExoCTK/contam_visibility/sossFieldSim.py]: And remove the commented lines in this function.

File: ExoCTK/contam_visibility/visibilityPA.py [hunk 9]
```diff
@@ -1,32 +1,30 @@
...
 import matplotlib
 matplotlib.use('Agg')
 import matplotlib.pyplot as plt
 import matplotlib.dates as mdates
-from matplotlib.ticker import AutoMinorLocator
 from matplotlib.ticker import MultipleLocator
-from astropy.io import ascii
-import os
-import pkg_resources
-import datetime
```
Comment (julesfowler) [inline ExoCTK/contam_visibility/visibilityPA.py]: And again with the imports.

File: ExoCTK/contam_visibility/visibilityPA.py [hunk 10]
```diff
@@ -36,190 +34,213 @@ def convert_ddmmss_to_float(astring):
...
-        
+
         # Top
-        err_y = np.concatenate([paMin[i0_top:i1_top+1],paMaxTmp[i0_top:i1_top+1][::-1]])
-        # err_x = np.concatenate([[d.timestamp() for d in gd[i0_top:i1_top+1]],[d.timestamp() for d in gd[i0_top:i1_top+1]][::-1]])
-        err_x = np.concatenate([gdMaskednum[i0_top:i1_top+1],gdMaskednum[i0_top:i1_top+1][::-1]])
+        err_y = np.concatenate([paMin[i0_top:i1_top+1],
+                                paMaxTmp[i0_top:i1_top+1][::-1]])
+        # err_x = np.concatenate([[d.timestamp() for d in gd[i0_top:i1_top+1]],
+        # [d.timestamp() for d in gd[i0_top:i1_top+1]][::-1]])
```
Comment (julesfowler) [inline ExoCTK/contam_visibility/visibilityPA.py]: RIP these commented lines.

File: ExoCTK/contam_visibility/sossFieldSim.py [hunk 11]
```diff
@@ -1,200 +1,223 @@
...
-    modelParam = readsav(os.path.join(idlsave_path,'modelsInfo.sav'),verbose=False) 
-    models     = modelParam['models']
-    modelPadX  = modelParam['modelpadx']
-    modelPadY  = modelParam['modelpady'] 
-    dimXmod    = modelParam['dimxmod']
-    dimYmod    = modelParam['dimymod']
-    jhMod      = modelParam['jhmod']
-    hkMod      = modelParam['hkmod']
-    teffMod    = modelParam['teffmod'] 
+    # cubeNameSuf = ''
```
Comment (jaymedina) [inline ExoCTK/contam_visibility/sossFieldSim.py]: What's the purpose of the comment on line 38?

File: ExoCTK/contam_visibility/sossFieldSim.py [hunk 12]
```diff
@@ -1,200 +1,223 @@
...
+    PAmin = 0  # instrument PA, degrees
     PAmax = 360
-    dPA = 1 # degrees
+    dPA = 1  # degrees
     # Set of IPA values to cover
     PAtab = np.arange(PAmin, PAmax, dPA)    # degrees
-    nPA    = len(PAtab)
+    nPA = len(PAtab)
 
     # dimX=256 #2048 #########now as argument, with default to 256
```
Comment (jaymedina) [inline ExoCTK/contam_visibility/sossFieldSim.py]: Delete

File: ExoCTK/contam_visibility/visibilityPA.py [hunk 13]
```diff
@@ -36,190 +34,213 @@ def convert_ddmmss_to_float(astring):
...
         fig.patch(err_x, err_y, color=color, fill_alpha=0.2, line_alpha=0)
 
         # Bottom
-        err_y = np.concatenate([paMinTmp[i0_bot:i1_bot+1],paMax[i0_bot:i1_bot+1][::-1]])
-        # err_x = np.concatenate([[d.timestamp() for d in gd[i0_bot:i1_bot+1]],[d.timestamp() for d in gd[i0_bot:i1_bot+1]][::-1]])
-        err_x = np.concatenate([gdMaskednum[i0_bot:i1_bot+1],gdMaskednum[i0_bot:i1_bot+1][::-1]])
+        err_y = np.concatenate([paMinTmp[i0_bot:i1_bot+1],
+                                paMax[i0_bot:i1_bot+1][::-1]])
+        # err_x = np.concatenate([[d.timestamp() for d in gd[i0_bot:i1_bot+1]],
+        # [d.timestamp() for d in gd[i0_bot:i1_bot+1]][::-1]])
```
Comment (jaymedina) [inline ExoCTK/contam_visibility/visibilityPA.py]: Delete lines 236, 237

File: ExoCTK/contam_visibility/resolve.py [hunk 14]
```diff
@@ -1,18 +1,27 @@
...
+
+    Parameters
+    ----------
+    targetName: str
+        The name of the target
+
+    Returns
+    -------
+    tuple
+        The ra and dec of the target"""
```
Comment (jaymedina) [inline ExoCTK/contam_visibility/resolve.py]: The """ that ends your docstring should be on its own.

File: ExoCTK/contam_visibility/generate_JH_HK_colors.py [hunk 15]
```diff
@@ -1,54 +1,78 @@
...
-    Kmags     = np.zeros(nModels)
+    nModels = 30
+    Jmags = np.zeros(nModels)
+    Hmags = np.zeros(nModels)
+    Kmags = np.zeros(nModels)
 
-    teff_list = [x for x in range(2800,5500+100,100)] + [5800] + [6000]
+    teff_list = [x for x in range(2800, 5500+100, 100)] + [5800] + [6000]
 
     # check if I did that write
```
Comment (jaymedina) [inline ExoCTK/contam_visibility/generate_JH_HK_colors.py]: *right 
🐱

File: ExoCTK/contam_visibility/math_extensionsx.py [hunk 16]
```diff
@@ -1,12 +1,11 @@
...
-""" 2018/06/26 Made PEP8 compliant and added
-    sind() and cosd() - Joe Filippazzo"""
-""" Version 1 August 23, 2010 RLH - Added OBLIQUITY."""
-""" Version 0 August 6, 2010 RLH  - Created"""
-
+library.
+2018/06/26 Made PEP8 compliant and added
+sind() and cosd() - Joe Filippazzo
+Version 1 August 23, 2010 RLH - Added OBLIQUITY.
+Version 0 August 6, 2010 RLH  - Created"""
```
Comment (jaymedina) [inline ExoCTK/contam_visibility/math_extensionsx.py]: The """ that ends this docstring should be in a line on its own.

File: ExoCTK/contam_visibility/quaternionx.py [hunk 17]
```diff
@@ -1,292 +1,328 @@
...
+    """Creates rotation quaternion about X axis, rotates a vector about
+    this axis"""
+    return Quaternion(Vector(sin(angle/2.), 0., 0.), cos(angle/2.))
+
 
 def QY(angle):
-   """Creates rotation quaternion about Y axis, rotates a vector about this axis"""
-   return Quaternion(Vector(0., math.sin(angle/2.), 0.), math.cos(angle/2.))
+    """Creates rotation quaternion about Y axis, rotates a vector about
+    this axis"""
```
Comment (jaymedina) [inline ExoCTK/contam_visibility/quaternionx.py]: """ that ends docstring should be on its own

File: ExoCTK/contam_visibility/quaternionx.py [hunk 18]
```diff
@@ -1,292 +1,328 @@
...
+    return Quaternion(Vector(0., sin(angle/2.), 0.), cos(angle/2.))
+
 
 def QZ(angle):
-   """Creates rotation quaternion about Z axis, rotates a vector about this axis"""
-   return Quaternion(Vector(0., 0., math.sin(angle/2.)), math.cos(angle/2.))
+    """Creates rotation quaternion about Z axis, rotates a vector about
+    this axis"""
```
Comment (jaymedina) [inline ExoCTK/contam_visibility/quaternionx.py]: """ that ends multiline docstring should be on its own

File: ExoCTK/contam_visibility/quaternionx.py [hunk 19]
```diff
@@ -1,292 +1,328 @@
...
-   return QZ(coord1)*QY(-coord2)*QX(-V3pa)
+    """Creates a rotation Q, going from the body frame to inertial"""
+    return QZ(coord1)*QY(-coord2)*QX(-V3pa)
+
 
 def Qmake_v2v3_2body(v2, v3):
-   """Creates a rotation Q, going from v2 and v3 in the body frame to inertial"""
-   return QY(v3)*QZ(-v2)
+    """Creates a rotation Q, going from v2 and v3 in the body frame to
+    inertial"""
```
Comment (jaymedina) [inline ExoCTK/contam_visibility/quaternionx.py]: """ that ends multiline docstring should be on its own

File: ExoCTK/contam_visibility/quaternionx.py [hunk 20]
```diff
@@ -1,292 +1,328 @@
...
-   r32 = 2.*(Q.q2*Q.q3 + Q.q1*Q.q4)
-   r33 = -Q.q1*Q.q1 - Q.q2*Q.q2 + Q.q3*Q.q3 + Q.q4*Q.q4
-   coord1 = math.atan2(r21, r11)
-   if coord1 < 0. : coord1 += PI2
-   coord2 = math.asin2(r31)  # use "safe" version of sine
-   pa  = math.atan2(-r32, r33)
-   if pa < 0. : pa += PI2
-   return (coord1, coord2, pa)
+    """Creates a angle tuple from Q, assuming body frame to inertial Q and
+    321 rotation sequence"""
```
Comment (jaymedina) [inline ExoCTK/contam_visibility/quaternionx.py]: General comment: 
Move the closing """ in a multiline docstring to its own line (lines 111, 125, and any others)

File: ExoCTK/contam_visibility/quaternionx.py [hunk 21]
```diff
@@ -1,292 +1,328 @@
...
     """Class to encapsulate matrix data and methods.
 
-   	A matrix is simply a list of lists that correspond to rows of the matrix.
-	This is just intended to handle simple multiplication and vector rotations.
-	For anything more advanced or computationally intensive, Python library routines
-	should be used."""
+    A matrix is simply a list of lists that correspond to rows of the matrix.
+    This is just intended to handle simple multiplication and vector rotations.
+    For anything more advanced or computationally intensive, Python library
+    routines should be used."""
```
Comment (jaymedina) [inline ExoCTK/contam_visibility/quaternionx.py]: closing """ needs to be moved down 1 line

File: ExoCTK/contam_visibility/quaternionx.py [hunk 22]
```diff
@@ -317,12 +353,12 @@ def element(self, row_index, col_index):
         """Returns an element of the matrix indexed by row and column.
 
         Indices begin with 0."""
```
Comment (jaymedina) [inline ExoCTK/contam_visibility/quaternionx.py]: this too

File: ExoCTK/contam_visibility/quaternionx.py [hunk 23]
```diff
@@ -338,316 +374,337 @@ def num_rows(self):
...
+
+        The degrees attribute should be True if the RA, DEC inputs are in
+        degrees. Otherwise radians is assumed.
+
+        The coordinates "ra" and "dec" may be used in all three systems.
+        Other names for coordinates in different frames may be defined for
+        clarity.
+
+        A CelestialVector is also an ordinary unit vector, with Cartesian
+        coordinates defined relative to the equatorial plane."""
```
Comment (jaymedina) [inline ExoCTK/contam_visibility/quaternionx.py]: """

File: ExoCTK/contam_visibility/quaternionx.py [hunk 24]
```diff
@@ -685,65 +745,71 @@ def transform_frame(self, new_frame):
...
+            result = self.rotate_about_axis(math2.OBLIQUITY, 'x')
+
         elif ((new_frame == 'gal') and (self.frame == 'eq')):
-            #Use formula from Wayne Kinzel's book, adjusted for J2000 coordinates.
-            b = math2.asin2(cos(self.dec) * cos(NGP.longitude) * cos(self.ra - NGP.latitude)\
-            + sin(self.dec) * sin(NGP.longitude))
+            # Use formula from Wayne Kinzel's book, adjusted for
+            # J2000 coordinates.
+            a = cos(self.dec)*cos(NGP.longitude)*cos(self.ra - NGP.latitude)
+            b = math2.asin2(a + sin(self.dec) * sin(NGP.longitude))
```
Comment (jaymedina) [inline ExoCTK/contam_visibility/quaternionx.py]: Should there be spaces around the `*` in line 763?

**Comment by jaymedina** [comment]
Most of my change requests involved moving the closing `"""` to its own line for your multiline docstrings. I pointed most of them out, but there were a few more, so I would recommend reviewing the docstrings before merging.

**Comment by hover2pi** [comment]
Thanks @julesfowler and @jaymedina ! Yeah, this was just so much code to fix, a lot slipped through the cracks. Some functions are black boxes so I skipped the docstrings (which is bad) and other things I just phoned in to save time (like better variable names) but this was a good second pass. I'll make the changes ASAP. Thanks!

File: ExoCTK/contam_visibility/ephemeris_old2x.py [hunk 25]
```diff
@@ -1,37 +1,33 @@
...
+R2D = 180. / math.pi  # radians to degrees
+PI2 = 2. * math.pi  # 2 math.pi
+MIN_SUN_ANGLE = 84.8 * D2R  # minimum Sun angle, in radians
+MAX_SUN_ANGLE = 135.0 * D2R  # maximum Sun angle, in radians
+SUN_ANGLE_PAD = 0.5 * D2R   # pad away from Sun angle limits
 
 obliquity_of_the_ecliptic = 23.439291  # At J2000 equinox
-obliquity_of_the_ecliptic *=  D2R
+obliquity_of_the_ecliptic *= D2R
 Qecl2eci = qx.QX(obliquity_of_the_ecliptic)
```
[the remark this answers is not in the document]
Comment (hover2pi) [reply ExoCTK/contam_visibility/ephemeris_old2x.py]: Yup. I simplified it.

File: ExoCTK/contam_visibility/ephemeris_old2x.py [hunk 26]
```diff
@@ -1,37 +1,33 @@
...
-    #08/03/2010 Got rid of degrees trig functions
-    #           Removed in_FOR, is_valid_att etc. as those are S/C dependent 
+    # 07/31/2008 Added functions for sun position
+    # 12/05/2008 Switched to spline fit.
+    # 07/29/2010 Added OP window calc
+    # 08/03/2010 Got rid of degrees trig functions
+    #           Removed in_FOR, is_valid_att etc. as those are S/C dependent
 
-    def __init__(self,afile,cnvrt=False):
+    def __init__(self, afile, cnvrt=False):
```
[the remark this answers is not in the document]
Comment (hover2pi) [reply ExoCTK/contam_visibility/ephemeris_old2x.py]: I changed the first but I'm leaving the `cnvrt` incase the kwarg is invoked in another module.

File: ExoCTK/contam_visibility/ephemeris_old2x.py [hunk 27]
```diff
@@ -71,192 +66,224 @@ def __init__(self,afile,cnvrt=False):
...
+        # # yp = spline(xa, ya, 0., 0.)
+        # Saving spline parameters
+        # self.xlistp = spline(self.datelist, self.xlist, 1.e31, 1.e31)
+        # self.ylistp = spline(self.datelist, self.ylist, 1.e31, 1.e31)
+        # self.zlistp = spline(self.datelist, self.zlist, 1.e31, 1.e31)
         fin.close()
-        #print len(self.datelist),len(self.xlist),len(self.ylist),len(self.zlist)
-        
+        # print len(self.datelist), len(self.xlist), len(self.ylist),
+        # len(self.zlist)
```
[the remark this answers is not in the document]
Comment (hover2pi) [reply ExoCTK/contam_visibility/ephemeris_old2x.py]: Version control!

**Comment by julesfowler** [comment]
@hover2pi Are you ready for this to be reviewed? Do u want me to look it over?

**Comment by hover2pi** [comment]
@julesfowler Not yet! I'm doing docstrings for the last module.

**Comment by hover2pi** [comment]
@julesfowler OK! It's good to go. Have at it!

**Comment by hover2pi** [comment]
99 problems but a doctoring ain't one.

**Pull request merged**